server/src/internalClusterTest/java/org/elasticsearch/search/SearchCancellationIT.java

```diff
@@ -180,6 +183,65 @@ public void testCancellationDuringFetchPhase() throws Exception {
         ensureSearchWasCancelled(searchResponse);
     }
 
+    public void testCancellationDuringAggregation() throws Exception {
+        List<ScriptedBlockPlugin> plugins = initBlockFactory();
+        indexTestData();
+
+        logger.info("Executing search");
+        ActionFuture<SearchResponse> searchResponse = client()
+            .prepareSearch("test")
+            .setQuery(matchAllQuery())
+            .addAggregation(
+                new TermsAggregationBuilder("test_agg")
+                    .script(new Script(
```

Comment:
Could you use a regular terms agg? If not probably worth a comment why. I spent a little while trying to figure out if the terms script created a second block or something before I went and read it.